src/gcp-algorithms/fastals.jl

```diff
@@ -0,0 +1,242 @@
+## Algorithm: FastALS
+
+"""
+    FastALS
+
+Fast Alternating Least Squares.
+Faster and more memory-efficient implementation of ALS from "Fast Alternating LS Algorithms
+for High Order CANDECOMP/PARAFAC Tensor Factorizations" by Phan et al.
+
+Algorithm parameters:
+
+- `maxiters::Int` : max number of iterations (default: `200`)
+
+"""
+Base.@kwdef struct FastALS <: AbstractAlgorithm
+    maxiters::Int = 200
+end
+
+function _gcp(
+    X::Array{TX,N},
+    r,
+    loss::GCPLosses.LeastSquaresLoss,
+    constraints::Tuple{},
+    algorithm::GCPAlgorithms.FastALS,
+    init,
+) where {TX<:Real,N}
+
+    # Initialization
+    M = deepcopy(init)
+
+    # Determine order of modes of MTTKRP to compute
+    Jns = [prod(size(X)[1:n]) for n in 1:N]
+    Kns = [prod(size(X)[n+1:end]) for n in 1:N]
+    Kn_minus_ones = [prod(size(X)[n:end]) for n in 1:N]
+    comp = Jns .<= Kn_minus_ones
+    n_star = maximum(map(x -> comp[x] ? x : 0, 1:N))
```

Comment:
Is this what these lines are trying to do?
```suggestion
    n_star = findlast(n -> Jns[n] <= Kn_minus_ones[n], 1:N)
```